addon/plugins/citation-plugin/utils/legal-documents.ts

```diff
@@ -0,0 +1,133 @@
+
+interface LegalDocumentArgs {
+  uri: string;
+  legislationTypeUri: Option<string>;
+  title: string | null;
+  publicationDate: string | null;
+  documentDate: string | null;
+  meta?: Record<string, unknown> | null;
+}
+
+export class LegalDocument {
+  uri: string;
+  legislationType: { label: string; value: string } | undefined;
+  title: string | null;
+  publicationDate: string | null;
+  documentDate: string | null;
+  meta: Record<string, unknown> | null;
+
+  constructor({
+    uri,
+    legislationTypeUri,
+    title,
+    publicationDate,
+    documentDate,
+    meta,
+  }: LegalDocumentArgs) {
+    this.uri = uri;
+    this.legislationType = LEGISLATION_TYPE_CONCEPTS.find(
+      (t) => t.value === legislationTypeUri,
+    );
+    this.title = title;
+    this.publicationDate = publicationDate;
+    this.documentDate = documentDate;
+    this.meta = meta ?? null;
+  }
+
+  get fullTitle() {
+    return `${this.legislationType?.label ?? ''} ${this.title ?? ''}`;
+  }
+}
+
+export const isBesluitLegalDocument = (legalDocument: LegalDocument) =>
+  isBesluitType(legalDocument.legislationType?.value ?? '');
+
+export const fetchLegalDocumentsCache = new Map<
+  string,
+  LegalDocumentsCollection
+>();
+
+export interface LegalDocumentsCollection {
+  totalCount: number;
+  legalDocuments: LegalDocument[];
+}
+
+export interface LegalDocumentsQueryFilter {
+  type: string;
+  documentDateFrom?: Option<string>;
+  documentDateTo?: Option<string>;
+  publicationDateFrom?: Option<string>;
+  publicationDateTo?: Option<string>;
+}
+
+export type LegalDocumentsQueryConfig = {
+  endpoint: string;
+  decisionsEndpoint?: string;
+  abortSignal?: AbortSignal;
+};
+
+export async function fetchLegalDocuments({
+  words,
+  filter,
+  pageNumber = 0,
+  pageSize = 5,
+  config,
+}: {
+  words: string[];
+  filter: LegalDocumentsQueryFilter;
+  pageNumber: number;
+  pageSize: number;
+  config: LegalDocumentsQueryConfig;
+}) {
+  //This is silly, but null != undefined, so we have to be careful and include the correct value here
+  //Also, reconstruct the whole filter object to always have the same ordering, hopefully.
+  filter = {
+    type: filter.type,
+    documentDateFrom: filter.documentDateFrom || null,
+    documentDateTo: filter.documentDateTo || null,
+    publicationDateFrom: filter.publicationDateFrom || null,
+    publicationDateTo: filter.publicationDateTo || null,
+  };
+  const stringArguments = JSON.stringify({
+    words,
+    filter,
+    pageNumber,
+    pageSize,
+  });
+  const results = fetchLegalDocumentsCache.get(stringArguments);
+
+  if (results) {
+    return results;
+  } else {
+    const shouldQueryPublicDecisions =
+      isBesluitType(filter.type) && config.decisionsEndpoint;
+
+    const fetchConfig = {
+      words,
+      filter,
+      pageNumber,
+      pageSize,
+      config: {
+        endpoint: shouldQueryPublicDecisions
+          ? (config.decisionsEndpoint as string)
+          : config.endpoint,
```

Comment:
if decisionsEndpoint is not provided, it will search for decisions on the other endpoint, which in our case will never return any. In a way this could make sense: a vendor might have a backend that provides everything (including decisions) in one endpoint.

However, I don't think this is a good idea:
- if this is the case with vendors, they can still define the same url for `decisionsEndpoint` and ` endpoint`
- I think it will always be separate, because all vendors will be using the same codex flemish codex url. The code for citation is too specific for this codex to really be used with anything else.
- vendors wanting to just use the codex (and no self-published decisions) will always see "besluit" in the dropdown, which will never return anything.

So I'd change the logic so that if `config.decisionsEndpoint` is not defined, the dropdown does not show `Besluit` as an option to choose from.